File: app/src/test/java/com/mgaetan89/showsrage/fragment/ShowsSectionFragment_IsShowValidTest.java
package com.mgaetan89.showsrage.fragment;

import com.google.gson.Gson;
import com.mgaetan89.showsrage.model.Show;
import com.mgaetan89.showsrage.network.SickRageApi;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;

import java.util.Arrays;
import java.util.Collection;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(Parameterized.class)
public class ShowsSectionFragment_IsShowValidTest {
	@Parameterized.Parameter(1)
	public boolean valid;

	@Parameterized.Parameter(0)
	public Show show;

	@Test
	public void isShowValid() {
		assertThat(ShowsSectionFragment.Companion.isShowValid(this.show)).isEqualTo(this.valid);
	}

	@Parameterized.Parameters
	public static Collection<Object[]> data() {
		Gson gson = SickRageApi.Companion.getGson();

		return Arrays.asList(new Object[][]{
				{null, false},
				{gson.fromJson("{}", Show.class), false},
				{gson.fromJson("{indexerid: 0}", Show.class), false},
				{gson.fromJson("{indexerid: 123}", Show.class), true},
		});
	}
}

File: app/src/test/kotlin/com/mgaetan89/showsrage/fragment/ShowsSectionFragment_IsShowValidTest.kt
package com.mgaetan89.showsrage.fragment

import com.mgaetan89.showsrage.model.Show
import com.mgaetan89.showsrage.network.SickRageApi
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized

@RunWith(Parameterized::class)
class ShowsSectionFragment_IsShowValidTest(val show: Show?, val valid: Boolean) {
    @Test
    fun isShowValid() {
        assertThat(ShowsSectionFragment.isShowValid(this.show)).isEqualTo(this.valid)
    }

    companion object {
        @JvmStatic
        @Parameterized.Parameters
        fun data(): Collection<Array<Any?>> {
            val gson = SickRageApi.gson

            return listOf(
                    arrayOf<Any?>(null, false),
                    arrayOf<Any?>(gson.fromJson("{}", Show::class.java), false),
                    arrayOf<Any?>(gson.fromJson("{indexerid: 0}", Show::class.java), false),
                    arrayOf<Any?>(gson.fromJson("{indexerid: 123}", Show::class.java), true)
            )
        }
    }
}
